src/TimeModeling/Types/GeometryStructure.jl

```diff
@@ -179,12 +179,13 @@ function Geometry(data::SegyIO.SeisBlock; key="source", segy_depth_key="")
     for j=1:nsrc
         traces = findall(src .== unique(src)[j])
         if key=="source"    # assume same source location for all traces within one shot record
-            xloc[j] = convert(gt, xloc_full[traces][1])
-            yloc[j] = convert(gt,yloc_full[traces][1])
-            zloc[j] = abs.(convert(gt,zloc_full[traces][1]))
+            xloc[j] = convert(gt, xloc_full[traces][1:1])
```

Comment:
It should contain `[1:1]` otherwise the `xloc_full[traces]` contains a repeated array, e.g. `[100.0, 100.0, 100.0]` but the `xloc[j]` of the source should really be `[100.0]`

Reply:
That wouldn't be an issue with your change to `gt`